Source/Core/InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.cpp

```diff
@@ -0,0 +1,355 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.h"
+
+#include <cassert>
+
+#include "Common/Config/Config.h"
+#include "Common/Flag.h"
+#include "Common/Logging/Log.h"
+#include "Common/MathUtil.h"
+#include "Common/Random.h"
+#include "Common/Thread.h"
+#include "Core/CoreTiming.h"
+#include "Core/HW/SystemTimers.h"
+#include "Core/HW/WiimoteEmu/Constants.h"
+#include "SFML/Network/SocketSelector.hpp"
+
+namespace ciface::CemuHookUDPServer
+{
+static constexpr char DEFAULT_SERVER_ADDRESS[] = "127.0.0.1";
+static constexpr u16 DEFAULT_SERVER_PORT = 26760;
+static constexpr auto SERVER_REREGISTER_INTERVAL = std::chrono::seconds{1};
+static constexpr auto SERVER_LISTPORTS_INTERVAL = std::chrono::seconds{1};
+static constexpr int TOUCH_X_AXIS_MAX = 1000;
+static constexpr int TOUCH_Y_AXIS_MAX = 500;
+
+namespace Settings
+{
+const Config::ConfigInfo<bool> SERVER_ENABLED{
+    {Config::System::CemuHookUdpServer, "Server", "Enabled"}, false};
+const Config::ConfigInfo<std::string> SERVER_ADDRESS{
+    {Config::System::CemuHookUdpServer, "Server", "IPAddress"}, DEFAULT_SERVER_ADDRESS};
+const Config::ConfigInfo<int> SERVER_PORT{
+    {Config::System::CemuHookUdpServer, "Server", "Port"}, DEFAULT_SERVER_PORT};
+}  // namespace Settings
+
+static bool s_server_enabled;
+static std::string s_server_address;
+static u16 s_server_port;
+static u32 s_client_uid;
+static std::chrono::steady_clock::time_point s_next_listports;
+static std::thread s_hotplug_thread;
+static Common::Flag s_hotplug_thread_running;
+std::mutex s_port_info_mutex;
+Proto::MessageType::PortInfo s_port_info[Proto::PORT_COUNT];
+
+static bool MachineIsLittleEndian()
```

Comment:
Feels like we have one of those already, somewhere in `Common`...

Reply:
I am unfortunately not able to locate it. If you are thinking of `Common/Swap.h` I can't see an equivalent function there.

Reply:
Indeed, we don't have a function like this. We've never really bothered trying to support big endian machines – there's lots of code spread out in Dolphin that assumes it's running on a little endian machine.

Reply:
OK, should I just remove the check?

Reply:
Personally I think it's fine to either keep it or remove it, but maybe someone else has a different opinion?

Reply:
Seems fine to remove it and maybe have a comment "assuming little endian host". If someone ever tries to support big endian hosts they can at least find this line with a search for "endian".

Reply:
Sidenote: If you want big endian support, relevant code should be using htole / htobe, and reversing functions instead of a test like above. C++20 defines a standardized way for these operations. Since there is no big endian support at this time, it's probably just safer to wait until C++20 can be adopted, unless an urgent need shows itself beforehand.